app/gui/view/component-browser/breadcrumbs/src/lib.rs

```diff
@@ -0,0 +1,478 @@
+            if let Some(last) = self.column_of_the_last_entry() {
+                if col < last {
+                    self.grid.select_entry(Some((row, col + 2)));
+                }
+            }
+        }
+    }
+
+    /// Clear the breadcrumbs list.
+    pub fn clear(&self) {
+        self.entries.borrow_mut().clear();
+        self.grey_out(None);
+        self.grid.resize_grid(1, 0);
+    }
+}
+
+
+// === Breadcrumb ===
+
+/// The breadcrumb type.
+#[derive(Debug, Clone, CloneRef, Default)]
+pub struct Breadcrumb(ImString);
+
+impl Breadcrumb {
+    /// Constructor.
+    pub fn new(label: &str) -> Self {
+        Self(ImString::new(label))
+    }
+}
+
+
+
+// ===========
+// === FRP ===
+// ===========
+
+ensogl_core::define_endpoints_2! {
+    Input {
+        /// Select a specific breadcrumb, greying out breadcrumbs after it.
+        select(BreadcrumbId),
+        /// Add a new breadcrumb after the currently selected one.
+        push(Breadcrumb),
+        /// Set a list of displayed breadcrumbs, rewriting any previously added breadcrumbs.
+        set_entries(Vec<Breadcrumb>),
+        /// Enable or disable displaying of the ellipsis icon at the end of the list.
+        show_ellipsis(bool),
+        /// Remove all breadcrumbs.
+        clear(),
+        /// Set the size of the visible portion of the breadcrumbs. The widget will crop the
+        /// breadcrumbs to this size and prioritize showing the right part of the list if it
+        /// can't fit in completely.
+        set_size(Vector2),
+        /// Move the selection to the previous (higher-level) breadcrumb in the list.
+        move_up(),
+        /// Move the selection to the next (lower-level) breadcrumb in the list.
+        move_down(),
+    }
+    Output {
+        /// Currently selected breadcrumb.
+        selected(BreadcrumbId)
+    }
+}
+
+
+
+/// ==============
+/// === Widget ===
+/// ==============
+
+#[derive(Debug, Clone, CloneRef, Deref)]
+pub struct Breadcrumbs {
+    #[deref]
+    widget: Widget<Model, Frp>,
+}
+
+impl Breadcrumbs {
+    /// Constructor.
+    pub fn new(app: &Application) -> Self {
+        let model = Rc::new(Model::new(app));
+        let display_object = model.display_object.clone_ref();
+        let frp = Frp::new();
+        let network = frp.network();
+        let input = &frp.private().input;
+        let out = &frp.private().output;
+        let grid = &model.grid;
+        let style = StyleWatchFrp::new(&app.display.default_scene.style_sheet);
+        let entries_height = style.get_number(theme::height);
+        let scroll_anim = Animation::new(network);
+        frp::extend! { network
+            init <- source_();
+            eval input.show_ellipsis((b) model.show_ellipsis(*b));
+            selected_grid_col <- grid.entry_selected.filter_map(|l| *l);
+            eval selected_grid_col([model]((_row, col)) {
+                model.grey_out(Some(col + 1));
+            });
```

Comment:
you dont need neither `[model]` here nor curly braces. Please clean similar places in other FRP places.

Reply:
This is an artifact of the older code changes. In some cases, our macro does require both `[model]` and curly braces.

Reply:
It should not. Could you please explain why it is needed here?

Reply:
It is not needed here, but it was required in my previous versions of the code. I changed these lines.